### PR TITLE
Add obsolete attributes to deprecated APIs and doc comments to methods

### PR DESCRIPTION
**Pull request by pakrym-stripe**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -4,7 +4,7 @@ update-version:
 	@perl -pi -e 's|<Version>[.\-\d\w]+</Version>|<Version>$(VERSION)</Version>|' src/Stripe.net/Stripe.net.csproj
 
 codegen-format:
-	dotnet format src/Stripe.net/Stripe.net.csproj
+	dotnet format src/Stripe.net/Stripe.net.csproj --severity warn
```
Comment (anniel-stripe) [inline Makefile:7]: why was this necessary?

Comment (pakrym-stripe) [reply Makefile:7]: Makes format slightly faster.

Comment (pakrym-stripe) [reply Makefile:7]: As it doesn't try to fix issues that won't result in an error.

**Review by anniel-stripe — APPROVED** [review]
